test/built-ins/Atomics/wait/null-or-false-timeout.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018 Amal Hussein.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-atomics.wait
+description: >
+  Null or true timeout arg should result in an +0 timeout
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  4.Let q be ? ToNumber(timeout).
+    ...
+    Null  Return +0.
+    Boolean	If argument is true, return 1. If argument is false, return +0.
+features: [ Atomics ]
+includes: [atomicsHelper.js]
+---*/
+
+var NUMREPORTS = 4; // Expected reports output from running agents
+
+function getReport() {
+  var r;
+  while ((r = $262.agent.getReport()) == null)
+    $262.agent.sleep(100);
+  return r;
+}
+
+$262.agent.start(
+  `
+$262.agent.receiveBroadcast(function (sab) {
+  var int32Array = new Int32Array(sab);
+  var start = Date.now();
+  $262.agent.report("A " + Atomics.wait(int32Array, 0, 0, null));  // null => +0
+  $262.agent.report(Date.now() - start);
+  $262.agent.leaving();
+})
+`);
+
+$262.agent.start(
+  `
+$262.agent.receiveBroadcast(function (sab) {
+  var int32Array = new Int32Array(sab);
+  var start = Date.now();
+  $262.agent.report("B " + Atomics.wait(int32Array, 0, 0, false));  // false => +0
+  $262.agent.report(Date.now() - start);
+  $262.agent.leaving();
+})
+`);
+
+var int32Array = new Int32Array(new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT));
+
+$262.agent.broadcast(int32Array.buffer);
+
+$262.agent.sleep(150);
+
+
+var sortedReports = [];
+for (var i = 0; i < NUMREPORTS; i++) {
+  sortedReports.push(getReport());
+}
+sortedReports.sort();
+
+assert.sameValue(sortedReports[2], 'A timed-out');
+assert.sameValue(sortedReports[3], 'B timed-out');
+assert(sortedReports[0] >= 0 && sortedReports[0] <= $ATOMICS_MAX_TIME_EPSILON, "timeout should be a min of 0ms and max of $ATOMICS_MAX_TIME_EPSILON");
```

Comment:
avoid DRY patterns in assertions, they can be split into different assertions. This helps clarity on what is exactly failing. e.g.: sortedReports[0] is < 0 or is > $ATOMICS_MAX_TIME_EPSILON.